### DISCUSSION TITLE
Allow creating an CanvasImage from an in-memory buffer

### DISCUSSION THREAD

**Discussion by pixsperdavid** [spectreconsole/spectre.console#243] [Ideas]
Hopefully self-explanatory. My use case for this is to allow CanvasImage to be used in single-file executables where images are embedded as binary resources rather than external files.  Happy to implement and submit a PR if approved.

**Comment by patriksvensson** [comment]
@impsnldavid Sorry for the late reply. Sure, sounds like a good idea!
